code/modules/power/cable.dm

```diff
@@ -415,3 +412,199 @@
 		logTheThing(LOG_STATION, user, "lays a cable[powered == 1 ? " (powered when connected)" : ""] at [log_loc(src)].")
 
 	return
+
+/// a cable spawner which can spawn multiple cables to connect to other cables around it.
+/obj/cablespawner
+	name = "power cable spawner"
+	icon = 'icons/obj/power_cond.dmi'
+	icon_state = "superstate"
+	layer = CABLE_LAYER
+	plane = PLANE_NOSHADOW_BELOW
+	color = "#DD0000"
+	// this would make it connect to the centre, for like terminals and whatnot
+	// has to be var edited because lazy
+	var/override_centre_connection = FALSE
+	var/cable_type = /obj/cable
+	/// the following 8 bit variable constants use bit flags to tell which 8 tiles need connections
+	/*
+	* bitflag of the tiles surrounding itself:
+	* i.e.(10)(9)(6)(5) 8  4  2  1 (normal bitflags) (used when you only need one direction)
+	* i.e. SW NW SE NE  W  E  S  N (corresponding directions)
+	*     128 64 32 16  8  4  2  1 (actual bitflags) (used for multiple directions)
+	* i think someone called it 8 bit directions once? idk
+	* anyway thats what cable_surr does
+	*/
+	var/const/SOUTHWEST_UNIQUE = 128
+	var/const/NORTHWEST_UNIQUE = 64
+	var/const/SOUTHEAST_UNIQUE = 32
+	var/const/NORTHEAST_UNIQUE = 16
```

Comment:
Regarding ZeWaka's comment I think it would be a good idea to make these `#define`s and move them to `_std/macros/dir.dm`.

However, a small request I have is to define them like this:
```
#define SOUTHWEST_UNIQUE (1<<6)
#define NORTHWEST_UNIQUE (1<<7)
#define SOUTHEAST_UNIQUE (1<<8)
#define NORTHEAST_UNIQUE (1<<9)
```
There are two reasons for that:
 - using `<<` makes it more obvious what's going on and less likely to make a typo in powers of 2
 - it seems like it'd make sense to start at (1<<4) because the 4 cardinal directions only use 0 to 3. However, technically BYOND has 6 cardinal directions. I don't think we ever use `UP` ( = 1<<4) and `DOWN` ( = 1<<5) but they do exist so I'd rather avoid collisions with those